### ISSUE TITLE
Access control on a @ApiSubresource()

### ISSUE THREAD

**Issue by augustinfla** [api-platform/api-platform#529]
Hello,

I do not find how to protect access to a subResource?
I would like to set up on the SubResource an object = user
How can we do it?

`
/**
 * @ORM\Table
 * @ORM\Entity
 * @ORM\EntityListeners({"AppBundle\EntityListener\UserEntityListener"})
 *
 * @ApiResource(attributes={
 *     "normalization_context"={"groups"={"user", "user-read"}},
 *     "denormalization_context"={"groups"={"user", "user-write"}}
 * }, collectionOperations={
 *     "post"={"method"="POST", "validation_groups"={"Default", "registration"}, "access_control"="is_granted('IS_AUTHENTICATED_ANONYMOUSLY')"},
 *     "get"={"method"="GET", "access_control"="is_granted('ROLE_ADMIN')"},
 * }, itemOperations={
 *     "get"={"method"="GET", "access_control"="is_granted('IS_AUTHENTICATED_ANONYMOUSLY')"},
 *     "put"={"method"="PUT", "access_control"="is_granted('ROLE_ADMIN') or object == user"},
 * })
 */
class User
{
     * @ORM\OneToMany(targetEntity="AppBundle\Entity\Address", mappedBy="user")
     * @ApiSubresource()
     */
    private $address;
`

When I want to display the following url it is necessary to return only the addresses of the connected user to secure the API

user/45/addresses

Thank

**Comment by mvigneron** [comment]
Hello,

Did you find a solution ? I have a same issue. 

access_control on subressources doesn't work...

I have configurated my entity like that : 

```
/**
 * @ApiResource(
 *     attributes={
 *      "normalization_context"={"groups"={"shop_client_bookmark_output"}},
 *      "denormalization_context"={"groups"={"shop_client_bookmark_input"}}
 *     },
 *     collectionOperations={
 *          "post"=
 *          {
 *              "method"="POST",
 *              "access_control"="has_role('ROLE_ADMIN') or (has_role('ROLE_CLIENT') and user and object.client==user)",
 *              "access_control_message"="You must be the owner to add this bookmark."
 *          },
 *          "api_clients_shop_bookmarks_get_subresource"=
 *          {
 *              "method"="GET",
 *              "access_control"="has_role('ROLE_OAUTH_AUTHENTICATED')",
 *              "access_control_message"="sub access",
*               "normalization_context"={"groups"={"client_shop_bookmarks"}}
 *          }
 *     },
 *     itemOperations={
 *      "get"={"method"="GET","access_control"="has_role('ROLE_OAUTH_AUTHENTICATED')"},
 *      "delete"={"method"="DELETE","access_control"="has_role('ROLE_ADMIN') or (has_role('ROLE_CLIENT') and user and object.client==user)"}
 *     }
 * )
```

**Comment by soyuka** [comment]
try: `subresourceOperations={}`

**Comment by mvigneron** [comment]
Hi, yes Just after I posted my message, I found this solution

```
@ApiResource(
 *     subresourceOperations={
 *          "api_clients_shop_bookmarks_get_subresource"= {
 *              "method"="GET",
 *              "access_control"="has_role('ROLE_OAUTH_AUTHENTICATED')"
 *          }
 *      }
 * )
```

Api docs is not exhaustive about that

Thanks

**Comment by soyuka** [comment]
> Api docs is not exhaustive about that

You're right! Please help us to improve it by contributing [here](https://github.com/api-platform/docs). Thanks!

**Comment by mvigneron** [comment]
Yes @soyuka , no problem,  when I will have a little more time, I will help you with pleasure ;-) 
For the moment, I have to develop my startup...

**Comment by liorchamla** [comment]
> api_clients_shop_bookmarks_get_subresource

Excuse me but I run the same problem (securing a SubResource) but I can't find why and how you used this string ?

Imagine I have a User entity with a subresource named $carts, what would be the string to securing it with subresourceOperations attribute ?

**Comment by jmgiaever** [comment]
As i have understood, you need to define the subresourceOperation on the class that is referenced, like this:

```php
/**
 * @ORM\...
 * @ApiResource()
 */
class Author {
	/**
	 * Subresource /authors/{id}/books
	 * @ORM\....
	 * @ApiSubresource
	 **/
	private $books;
}

/**
 * @ORM\...
 * @ApiResource({
 *	subresourceOperations = {
 *		"api_authors_books_get_subresource" = {
 *			"access_control" = "<rule(s)>"
 * 		}
 * 	}
 * })
 */
class Book {
 	//...
}
```

The scope is built on the request:
get authors (with id) books as subresource
